Firestore/Source/Remote/FSTRemoteEvent.h

```diff
@@ -165,34 +122,38 @@ typedef NS_ENUM(NSUInteger, FSTCurrentStatusUpdate) {
 
 - (instancetype)
 initWithSnapshotVersion:(firebase::firestore::model::SnapshotVersion)snapshotVersion
-          targetChanges:(NSMutableDictionary<NSNumber *, FSTTargetChange *> *)targetChanges
+          targetChanges:(std::unordered_map<FSTTargetID, FSTTargetChange *>)targetChanges
+       targetMismatches:(std::unordered_set<FSTTargetID>)targetMismatches
         documentUpdates:
-            (std::map<firebase::firestore::model::DocumentKey, FSTMaybeDocument *>)documentUpdates
+            (std::unordered_map<firebase::firestore::model::DocumentKey,
```

Comment:
Optional: consider adding a `using` alias for this type so you don't have to repeat it throughout.

```c++
using DocumentUpdateMap = std::unordered_map<
    firebase::firestore::model::DocumentKey,
    FSTMaybeDocument *,
    firebase::firestore::model::DocumentKeyHash>;
```

Reply:
I think this is a good optimization, but I opted not to do this, as this conflicts somewhat with the MaybeDocumentMap. Using the explicit type makes it really clear what is going on here.